test/expr/delayed-ident/member_chains.swift

```diff
@@ -0,0 +1,211 @@
+        ImplicitMembers()
+    }
+    
+    subscript(arg: Void) -> ImplicitMembers { ImplicitMembers() }
+    subscript(optional arg: Void) -> ImplicitMembers? { ImplicitMembers() }
+    subscript(func arg: Void) -> (() -> ImplicitMembers) { { ImplicitMembers() } }
+    subscript(funcOptional arg: Void) -> (() -> ImplicitMembers?) { { ImplicitMembers() } }
+    subscript(optionalFunc arg: Void) -> (() -> ImplicitMembers)? { { ImplicitMembers() } }
+}
+
+let _: ImplicitMembers = .implicit
+let _: ImplicitMembers? = .implicit
+let _: ImplicitMembers? = .optional
+
+let _: ImplicitMembers = .implicit.another.another
+let _: ImplicitMembers = .createImplicit().another.another
+let _: ImplicitMembers = .init().another.another
+
+let _: ImplicitMembers = .implicit.getAnother().another
+let _: ImplicitMembers = .createImplicit().getAnother().another
+let _: ImplicitMembers = .init().getAnother().another
+
+let _: ImplicitMembers = .implicit.getAnother(arg: 0).another
+let _: ImplicitMembers = .createImplicit().getAnother(arg: 0).another
+let _: ImplicitMembers = .init().getAnother(arg: 0).another
+
+let _: ImplicitMembers = .implicit.another.getAnother()
+let _: ImplicitMembers = .createImplicit().another.getAnother()
+let _: ImplicitMembers = .init().another.getAnother()
+
+let _: ImplicitMembers = .implicit.another.getAnother(arg: 0)
+let _: ImplicitMembers = .createImplicit().another.getAnother(arg: 0)
+let _: ImplicitMembers = .init().another.getAnother(arg: 0)
+
+let _: ImplicitMembers = .implicit.getAnother().getAnother(arg: 0)
+let _: ImplicitMembers = .createImplicit().getAnother().getAnother(arg: 0)
+let _: ImplicitMembers = .init().getAnother().getAnother(arg: 0)
+
+let _: ImplicitMembers = .implicit.getAnother().getAnother(arg: 0).another
+let _: ImplicitMembers = .createImplicit().getAnother().getAnother(arg: 0).another
+let _: ImplicitMembers = .init().getAnother().getAnother(arg: 0).another
+
+let _: ImplicitMembers = .implicit.another.getAnother().getAnother(arg: 0)
+let _: ImplicitMembers = .createImplicit().another.getAnother().getAnother(arg: 0)
+let _: ImplicitMembers = .init().another.getAnother().getAnother(arg: 0)
+
+let _: ImplicitMembers = .implicit.another.another.another.another.another
+let _: ImplicitMembers = .implicit.getAnother().getAnother().getAnother().getAnother().getAnother()
+let _: ImplicitMembers = .implicit.getAnother(arg: 0).getAnother(arg: 0).getAnother(arg: 0).getAnother(arg: 0).getAnother(arg: 0)
+
+let _: ImplicitMembers = .optional!
+let _: ImplicitMembers = .optional!.another
+let _: ImplicitMembers = .createOptional()!.another
+let _: ImplicitMembers = .optional!.anotherOptional!
+let _: ImplicitMembers = .createOptional()!.anotherOptional!
+let _: ImplicitMembers = .optional!.getAnotherOptional()!
+let _: ImplicitMembers = .createOptional()!.getAnotherOptional()!
+
+let _: ImplicitMembers = .other.implicit
+let _: ImplicitMembers = .implicit.anotherOther.implicit
```

Comment:
I think that it may be useful to have tests like `let _: ImplicitMembers = .implicit.anotherOther.nonDeclaredMember` to ensure that the components are being resolved correctly and this diagnoses that the type of `anotherOther` doesn't have a member named `nonDeclaredMember`.

Not sure, but maybe some way to test which base type is being inferred for each component down the chain, 
to ensure that they are being resolved correctly
Does this make sense? cc @xedin

Reply:
Yep, both of those make sense! For the second one, what tooling is available to test the inference of different subexpressions like that?

Reply:
I think there is some way to test this but don't know exactly how ...  @xedin or @CodaFi should know :)